src/controllers/user.js

```diff
@@ -106,11 +87,48 @@ const getUserInfoById = async (req, res) => {
   res.json(response({ data }));
 };
 
+/**
+ * Update user roles.
+ *
+ * @param {import('express').Request} req Express request object.
+ * @param {import('express').Response} res Express response object.
+ */
+const updateUserRoles = async (req, res) => {
+  const { roles } = req.body;
+  const userId = req.user.id;
+
+  await UserRoles.deleteByCondition({ userId });
+
+  if (roles.length) {
+    await UserRoles.createAll(roles.map((role) => ({ userId, roleId: role })));
+  }
+
+  res.json(response());
+};
+
+/**
+ * get user roles.
+ *
+ * @param {import('express').Request} req Express request object.
+ * @param {import('express').Response} res Express response object.
+ */
+const getUserRoles = async (req, res) => {
+  const userId = req.user.id;
+
+  const data = await UserRoles.getAllByCondition(
+    { userId },
+    { attributes: ['roleId'], include: [{ model: Roles, as: 'role', attributes: ['role'] }] }
+  );
```

Comment:
why not returning an array of roles?
```suggestion
  const data = (
    await User.getOne(userId, {
      include: {
        model: Roles,
        as: 'roles',
        attributes: ['id', 'role'],
        through: { attributes: [] },
      },
    })
  ).roles;
```